### PR TITLE
DEP: remove spatial.distance.matching

### LINKED ISSUE #15741: DEP: remove spatial.distance.matching
It's been deprecated since https://github.com/scipy/scipy/pull/7630

### PR DESCRIPTION
**Pull request by jyuv**
#### Reference issue

Closes #15741 

#### What does this implement/fix?

Remove matching from spatial.distance. Removed it also from test_distance's tests

### CODE REVIEW

File: scipy/spatial/distance.py [hunk 1]
```diff
@@ -2156,10 +2144,6 @@ def pdist(X, metric='euclidean', *, out=None, **kwargs):
         Computes the Yule distance between each pair of boolean
         vectors. (see yule function documentation)
 
-    16. ``Y = pdist(X, 'matching')``
-
-        Synonym for 'hamming'.
-
     17. ``Y = pdist(X, 'dice')``
```
Comment (peterbell10) [inline scipy/spatial/distance.py:2147]: Can you update the subsequent numbers.

Comment (jyuv) [reply scipy/spatial/distance.py:2147]: Hi, sorry for the delay. I just did

Comment (rgommers) [reply scipy/spatial/distance.py:2147]: @jyuv can you please have another look? You did it in another place, but here the numbering still goes from 15 to 17

**Comment by j-bowhay** [comment]
superseded by #16067

**Comment by j-bowhay** [comment]
superseded by gh-15741

**Comment by rgommers** [comment]
gh-15741 was merged, so closing this. Thank you for the contribution nevertheless @jyuv

**Pull request closed without merging**